moto/organizations/models.py

```diff
@@ -821,5 +821,33 @@ def disable_policy_type(self, **kwargs):
 
         return dict(Root=root.describe())
 
+    def detach_policy(self, **kwargs):
+        policy = self.get_policy_by_id(kwargs["PolicyId"])
+        if re.compile(utils.ROOT_ID_REGEX).match(kwargs["TargetId"]) or re.compile(
```

Comment:
This can be made more readable by extracting these into local/class variables.
Can we rewrite these regex-statements to say `root_id_regex.match(target_id)`?

Whether `root_id_regex` is defined doesn't matter much IMO, as long as it's only done once, instead of every call to `detach_policy`

Reply:
@bblommers I have updated based on your comments. I hope this satisfies the requirements.